### PR TITLE
Feature/cax 1 mtpdc@77 split test data

### PR DESCRIPTION
**Pull request by algattik**
Load in each partition, only test data for that partition.

Successful PRS Deploy run: https://github.com/catenax/tractusx/actions/runs/1424839200

### CODE REVIEW

**Review by marcgs — APPROVED** [review]

File: coreservices/partsrelationshipservice/cd/test-data/generate-test-data.sh [hunk 1]
```diff
@@ -16,15 +19,24 @@ echo 'COPY public.part_relationship (oneidmanufacturer, objectidmanufacturer, pa
 jq -r '(now | strftime("%Y-%m-%dT%H:%M:%S%z")) as $n | .[].relationships | .[].relationship | [.child.oneIDManufacturer, .child.objectIDManufacturer, .parent
 .oneIDManufacturer, .parent.objectIDManufacturer, "78F4BB1B-2EBB-418C-9C16-3E74BACCBEAC", $n] | @csv' "PartRelationshipUpdateList.json" | sort | uniq
 echo '\.'
+echo 'DELETE FROM public.part_relationship WHERE parent_oneidmanufacturer NOT IN ('
+jq -r '.partitions[] | select (.key=="'$partition$'") | .OneIDs[] | "\'"+.+"\',"' ../../dataspace-partitions.json
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/cd/test-data/generate-test-data.sh:23]: Can this be part of a variable and then reused , rather than duplicated for 3 tables ?

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java [hunk 2]
```diff
@@ -65,9 +65,14 @@
     public PartRelationshipsWithInfos toPartRelationshipsWithInfos(final Collection<PartRelationshipEntity> source, final Collection<PartIdEntityPart> allPartIds, final Collection<PartAttributeEntity> typeNames, final Collection<PartAspectEntity> aspects) {
         final var attributeIndex = typeNames.stream().collect(Collectors.groupingBy(t -> t.getKey().getPartId()));
         final var aspectIndex = aspects.stream().collect(Collectors.groupingBy(t -> t.getKey().getPartId()));
+
+        // Remove ids without any attribute information
+        // This case happens when a children node belongs to a different partition than the parent node
+        final Collection<PartIdEntityPart> sourceForPartInfos = allPartIds.stream().filter(attributeIndex::containsKey).collect(Collectors.toList());
+
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java:72]: is this change needed to split test data or committed accidently ? Asking this because PR title and description only contains information about split test data.

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java [hunk 3]
```diff
@@ -65,9 +65,14 @@
     public PartRelationshipsWithInfos toPartRelationshipsWithInfos(final Collection<PartRelationshipEntity> source, final Collection<PartIdEntityPart> allPartIds, final Collection<PartAttributeEntity> typeNames, final Collection<PartAspectEntity> aspects) {
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java:65]: If we do this changes here then can we also this update parameter name ? 
```suggestion
    public PartRelationshipsWithInfos toPartRelationshipsWithInfos(final Collection<PartRelationshipEntity> source, final Collection<PartIdEntityPart> allPartIds, final Collection<PartAttributeEntity> attributes, final Collection<PartAspectEntity> aspects) {
```

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java (hunk 2 above)
Comment (algattik) [reply coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java:72]: Yes there was a side effect (as indicated in the comment), where we would return spurious empty collections

**Review by Izzzu — APPROVED** [review]

**Review by cpeeyush — APPROVED** [review]

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java (hunk 2 above)
Comment (marcgs) [reply coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/mappers/PartRelationshipEntityListToDtoMapper.java:72]: Partitioning data lead to some nodes being present as leave nodes in the tree (as children of some other parent), but having no aspect nor attribute info within the partition as they belonged to another partition. So it was actually a small change in functionality required by the distributed setup.
